test/unit/local/steps/filter_ignored.js

```diff
@@ -0,0 +1,130 @@
+const Buffer = require('../../../../core/local/steps/buffer')
+const filterIgnored = require('../../../../core/local/steps/filter_ignored')
+
+describe('local/steps/filter_ignored', () => {
+  const builders = new Builders()
+
+  let ignore
+  let buffer
+
+  beforeEach(() => {
+    buffer = new Buffer()
+
+    const patterns = ['*.bck', 'tmp/', 'folder/']
+    ignore = new Ignore(patterns)
+  })
+
+  context('without any batches of events', () => {
+    it('does not throw any errors', () => {
+      should(() => filterIgnored(buffer, { ignore })).not.throw()
+    })
+
+    it('returns a buffer', () => {
+      should(filterIgnored(buffer, { ignore })).be.an.instanceof(Buffer)
+    })
+  })
+
+  context('with 1 batch of file & folder events', () => {
+    beforeEach(() => {
+      buffer.push([
+        builders.event().path('ignored.bck').kind('file').build(),
+        builders.event().path('notIgnored.txt').kind('file').build(),
+        builders.event().path('tmp/ignored.txt').kind('file').build(),
+        builders.event().path('notIgnored').kind('directory').build(),
+        builders.event().path('data/notIgnored.txt').kind('file').build()
+      ])
+    })
+
+    it('returns a buffer without events for filtered paths', async () => {
+      buffer = filterIgnored(buffer, { ignore })
+
+      const filteredEvents = await buffer.pop()
+      should(filteredEvents.map(event => event.path)).not.containDeep([
+        'ignored.bck',
+        'tmp/ignored.txt'
+      ])
+    })
+
+    it('keeps the order of unfiltered events', async () => {
+      buffer = filterIgnored(buffer, { ignore })
+
+      const filteredEvents = await buffer.pop()
+      should(filteredEvents).have.length(3)
+      should(filteredEvents.map(event => event.path)).containDeepOrdered([
+        'notIgnored.txt',
+        'notIgnored',
+        'data/notIgnored.txt'
+      ])
+    })
+  })
+
+  context('with multiple batches of file & folder events', () => {
+    beforeEach(() => {
+      buffer.push([
+        builders.event().path('ignored1.bck').kind('file').build(),
+        builders.event().path('notIgnored1.txt').kind('file').build()
+      ])
+      buffer.push([
+        builders.event().path('ignored2.bck').kind('file').build(),
+        builders.event().path('notIgnored2.txt').kind('file').build()
+      ])
+    })
+
+    it('returns a buffer without events for filtered paths', async () => {
+      let filteredEvents
+
+      buffer = filterIgnored(buffer, { ignore })
+
+      filteredEvents = await buffer.pop()
+      should(filteredEvents.map(event => event.path)).not.containDeep([
+        'ignored1.bck',
+        'ignored2.bck'
+      ])
+      filteredEvents = await buffer.pop()
+      should(filteredEvents.map(event => event.path)).not.containDeep([
+        'ignored1.bck',
+        'ignored2.bck'
+      ])
+    })
+
+    it('keeps the order of batches with unfiltered events', async () => {
+      let filteredEvents
```

Comment:
Aren't those `filteredEvents` actually the *unfiltered* ones according to the test description above?
Same question in all other tests.

Reply:
The language is tricky here as filtering it just the action of separating elements based on some criteria. It doesn't tell us anything on what's kept.
To avoid confusion I usually use the `filter out / filtered out` expression when I'm talking about the elements I'm not keeping ;-)

Reply:
In that case, `unfiltered` is really misleading: it should mean the events that have not been given to the filter, but we don't test the order of batches for those events.

Reply:
`relevantEvents`?
`eventsToProcess`?

Reply:
`relevantEvents` looks good to me